src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -752,6 +760,16 @@ inline sycl::event rotg(const char *func_name, Func func, sycl::queue &queue, T1
                         T1 *s, const std::vector<sycl::event> &dependencies) {
     using cuDataType1 = typename CudaEquivalentType<T1>::Type;
     using cuDataType2 = typename CudaEquivalentType<T2>::Type;
+    auto ctx = queue.get_context();
+    bool results_on_device = sycl::get_pointer_type(a, ctx) == sycl::usm::alloc::device;
+    if (results_on_device) {
+        if (sycl::get_pointer_type(b, ctx) == sycl::usm::alloc::unknown ||
```

Comment:
`unknown` is the only type that is not device-accessible. We could remove the check entirely, users will get a segfault instead of this error message.